From 971d081cd0653c0d6e89d2e26b4e56f3365f3485 Mon Sep 17 00:00:00 2001
From: Robert Chiras <robert.chiras@nxp.com>
Date: Wed, 29 Aug 2018 11:45:25 +0300
Subject: [PATCH 4482/5242] MLK-19243: arm64: dts: fsl-imx8mq-evk: fix adv7535
 i2c master

commit  b1ce455ee4e70586006399e8af3e99178b02b282 from
https://source.codeaurora.org/external/imx/linux-imx.git

Since the rev. B4 of the 8MQ EVK board, the i2c of the DSI was moved
from i2c1 to i2c3. Since ADV7535 needs these i2c lines, it was the only
one affected by this change.

Signed-off-by: Robert Chiras <robert.chiras@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../arm64/boot/dts/freescale/fsl-imx8mq-evk-b3.dts |   19 +++++++++++
 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts   |   36 ++++++++++----------
 2 files changed, 37 insertions(+), 18 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-b3.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-b3.dts
index 9ad6754..adc18e0 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-b3.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-b3.dts
@@ -16,6 +16,7 @@
 
 /delete-node/ &ov5640_mipi;
 /delete-node/ &ov5640_mipi2;
+/delete-node/ &adv_bridge;
 
 &i2c1 {
 	ov5640_mipi: ov5640_mipi@1c {
@@ -63,6 +64,24 @@
 			};
 		};
 	};
+
+	adv_bridge: adv7535@3d {
+		compatible = "adi,adv7533";
+		reg = <0x3d>;
+		adi,addr-cec = <0x3b>;
+		adi,dsi-lanes = <4>;
+		pinctrl-0 = <&pinctrl_i2c1_dsi_ts_int>;
+		interrupt-parent = <&gpio5>;
+		interrupts = <7 IRQ_TYPE_LEVEL_LOW>;
+
+		status = "disabled";
+
+		port {
+			adv7535_in: endpoint {
+				remote-endpoint = <&mipi_dsi_bridge_adv>;
+			};
+		};
+	};
 };
 
 &i2c2 {
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
index b4bfc3b..e4651f4 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
@@ -583,24 +583,6 @@
 		};
 	};
 
-	adv_bridge: adv7535@3d {
-		compatible = "adi,adv7533";
-		reg = <0x3d>;
-		adi,addr-cec = <0x3b>;
-		adi,dsi-lanes = <4>;
-		pinctrl-0 = <&pinctrl_i2c1_dsi_ts_int>;
-		interrupt-parent = <&gpio5>;
-		interrupts = <7 IRQ_TYPE_LEVEL_LOW>;
-
-		status = "disabled";
-
-		port {
-			adv7535_in: endpoint {
-				remote-endpoint = <&mipi_dsi_bridge_adv>;
-			};
-		};
-	};
-
 	dsi_touch: synaptics_dsx_ts@20 {
 		compatible = "synaptics_dsx";
 		reg = <0x20>;
@@ -670,6 +652,24 @@
 		reg = <0x11>;
 		ak4497,pdn-gpio = <&gpio3 16 GPIO_ACTIVE_HIGH>;
 	};
+
+	adv_bridge: adv7535@3d {
+		compatible = "adi,adv7533";
+		reg = <0x3d>;
+		adi,addr-cec = <0x3b>;
+		adi,dsi-lanes = <4>;
+		pinctrl-0 = <&pinctrl_i2c1_dsi_ts_int>;
+		interrupt-parent = <&gpio5>;
+		interrupts = <7 IRQ_TYPE_LEVEL_LOW>;
+
+		status = "disabled";
+
+		port {
+			adv7535_in: endpoint {
+				remote-endpoint = <&mipi_dsi_bridge_adv>;
+			};
+		};
+	};
 };
 
 &pcie0{
-- 
1.7.9.5

